src/components/currency/Currency.js

```diff
@@ -7,24 +7,20 @@ import {NumberComponent} from '../number/Number';
 export class CurrencyComponent extends NumberComponent {
   constructor(component, options, data) {
     super(component, options, data);
-
     this.currency = this.component.currency || 'USD';
-    this.decimalLimit = this.component.decimalLimit || 2;
-
+    this.decimalLimit = options.decimalLimit || this.component.decimalLimit || 2;
     // Get the prefix and suffix from the localized string.
-    const regex = `(.*)?100(${this.decimalSeparator === '.'
-      ? '\\.'
-      : this.decimalSeparator}0{${this.decimalLimit}})?(.*)?`;
-    const parts = (100).toLocaleString(options.language, this.getFormatOptions()).match(new RegExp(regex));
+    const regex = `(.*)?100(${this.decimalSeparator}0{${this.decimalLimit}})?(.*)?`;
```

Comment:
What was the purpose for these changes? The existing code came from standard libraries of how to locate the decimal and thousand separators. Is this going to cause any side effects?

Reply:
But that Previous regex gave me a wrong suffix when the decimalSeparator was ",". I changed it to this and works correctly.